src/shared/components/cancerSummary/CancerSummary.tsx

```diff
@@ -0,0 +1,448 @@
+    mutated:number;
+    amplified:number;
+    deleted:number;
+    multiple:number;
+}
+
+export interface IBarGraphDatasets {
+    label: string;
+    totalCases: number;
+    altTotalPercent: number;
+    total: number;
+    count: number;
+    percent: number;
+    data: number[];
+    backgroundColor: string;
+}
+
+export interface IBarGraphData {
+    labels: string[];
+    datasets: IBarGraphDatasets[];
+}
+
+export interface ICancerAlterationData extends ICancerAlterationPlotData {
+    total:number;
+}
+
+interface IGeneAltertionData {
+    [gene:string]: {
+        [cancerType:string]:ICancerAlterationData
+    };
+}
+
+interface ICancerSummaryContentProps {
+    data: {
+        [cancerType:string]:ICancerAlterationData
+    };
+}
+
+export interface ICancerSummaryContentState {
+    yAxis: 'alt-freq' | 'abs-count';
+    xAxis: 'y-axis' | 'can-types';
+    altSamples: {value: number; max: number};
+    totalSamples: {value: number; max: number};
+    showGenomicAlt: boolean;
+    cancerType: string;
+    datasets?: {
+        data: IBarGraphDatasets[];
+        label: string;
+        sortCount: number;
+        sortBy: string;
+    }[];
+}
+
+@observer
+export class CancerSummaryContent extends React.Component<ICancerSummaryContentProps, ICancerSummaryContentState> {
+
+
+    private inputYAxisEl:any;
+    private inputXAxisEl:any;
+    @observable private pngAnchor:any = '';
+    @observable private pdfAnchor = '';
+
+    @observable private showControls=false;
+    @observable private showGenomicAlt = true;
+    @observable private yAxis: 'alt-freq' | 'abs-count' = 'alt-freq';
+    @observable private xAxis: 'y-axis' | 'can-types' = 'y-axis';
+    @observable private multiSelectValue:string = this.cancerTypes[0].value;
+    @observable private multiSelect:any = {
+        options: this.cancerTypes,
+        value: this.multiSelectValue
+    };
+    @observable private altCasesValue = 0;
+    @observable private totalCasesValue = 0;
+
+    constructor(props:ICancerSummaryContentProps) {
+        super(props);
+
+        this.handleYAxisChange = this.handleYAxisChange.bind(this);
+        this.handleXAxisChange = this.handleXAxisChange.bind(this);
+        this.handleAltSliderChange = this.handleAltSliderChange.bind(this);
+        this.handleGenomicCheckboxChange = this.handleGenomicCheckboxChange.bind(this);
+        this.handleSelectChange = this.handleSelectChange.bind(this);
+        this.handleTotalSliderChange = this.handleTotalSliderChange.bind(this);
+        this.toggleShowControls = this.toggleShowControls.bind(this);
+        this.setPngAnchor = this.setPngAnchor.bind(this);
+        this.setPdfAnchor = this.setPdfAnchor.bind(this);
+        this.downloadPdf = this.downloadPdf.bind(this);
+        this.resetSliders = this.resetSliders.bind(this);
+    }
+
+    @computed get datasets() {
+        const {data} = this.props;
+        const {yAxis, showGenomicAlt, multiSelectValue:msv} = this;
+        return _.without(
```

Comment:
code this deeply nested is very hard to read.  instead of using _.without, I think you should use _.reduce to either discard items or include the transformed item.